### PR TITLE
Adding the rejection policy.

### LINKED ISSUE #115: Implement rejection policy
Conceptually the pipeline has the following parts:
- Processing a file (generate flags)
     - Automated processing
     - QC
-  Loading a pylossless output (reject epochs, channels, and ICs based on the flags)

The rejection should be based on an established "policy" which describes how the flags should be mapped to rejections. We should have an existing default policy that can be changed using a config file. We should probably create a Policy class to manage this. 

E.g. of default policy could be :
- reject automated components not brain or other with probability larger than 0.3 (from the automated process)
- reject manual labels (from the QC process).

### PR DESCRIPTION
**Pull request by christian-oreilly**
Fix #115

### CODE REVIEW

**Comment by christian-oreilly** [comment]
I am happy with that. The functionalities it provides are basic, but it helps structure PyLossless. @scott-huberty You can review whenever convenient.

**Comment by christian-oreilly** [comment]
That is strange. All the stuff about deleting `_check_sfreq` (actually, all the modifs in pipeline.py) is not something I made so I am not so sure how it got there. I created the branch from the main and completed the whole thing within the span of an evening... I am not sure how it interfered with these changes... Do these changes ring a bell for you @scott-huberty ?

**Comment by christian-oreilly** [comment]
Nevermind... I just reverted the commit for this file. Strange though...

**Comment by christian-oreilly** [comment]
@scott-huberty The test that fails is currently in pylossless/tests/test_simulated.py .  I did not change this file. Maybe it is related to this weird commit that I needed to revert? Would you mind having a look at it when you have some time? There is no rush for me.

**Comment by christian-oreilly** [comment]
@scott-huberty These CI issues are unrelated to this PR. Since you do not have time to look at it now and it has been blocking this PR for a while, I propose we merge this one in regardless so that we can use this functionality.

File: pylossless/tests/test_rejection.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    pipeline.run_with_raw(pipeline.raw)
+
+    rejection_config_path = Path("test_rejection_config.yaml")
+    rejection_config = ll.Config()
+    rejection_config["ch_flags_to_reject"] = ['ch_sd', 'low_r']
+    rejection_config.save(rejection_config_path)
+
+    raw = ll.RejectionPolicy(rejection_config_path).apply(pipeline)
+    rejection_config_path.unlink()
+
```
Comment (scott-huberty) [inline pylossless/tests/test_rejection.py:27]: The typo I pointed out in `rejection.py` caused this class to not behave as expected. We should add tests to check that the code behaves as expected and not just that it runs without raising errors. something like:

```
 if clean_ch_mode == None:
    assert_array_equal(flagged_chs, raw.info["bads"])
elif clean_ch_mode == "drop":
    not_in_ch_names = [ch for ch in flagged_chs if ch not in raw.ch_names]
    assert_array_equal(not_in_ch_names, flagged_chs)
elif clean_ch_mode == "interpolate":
    # interpolate_bads will drop ch_names from raw.info["bads"]
    # so to make sure that the channel was interpolated, lets check it
    assert_array_equal(flagged_chs, raw.info["bads"])
```

File: pylossless/rejection.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        self.epoch_flags_to_reject = []
+        self.ic_flags_to_reject = ['muscle', 'heart', 'ch_noise', 'eye']
+
+        self.ic_rejection_threshold = 0.3
+
+        # Can be:
+        #   "": Do nothing aside from adding them to raw.info["bads"]
+        #   "drop": Drop bad channels.
+        #   "interpolate": Interpolate bad channels.
+        self.clean_ch_mode = ""
```
Comment (scott-huberty) [inline pylossless/rejection.py:35]: can we change the default to `None`? That seems more idiomatic to me. 

Can we allow this to be set in the constructer? i.e. `clean_ch_mode=None` by default, and `self.clean_ch_mode=clean_ch_mode`

**Comment by scott-huberty** [comment]
> @scott-huberty These CI issues are unrelated to this PR. Since you do not have time to look at it now and it has been blocking this PR for a while, I propose we merge this one in regardless so that we can use this functionality.

Thanks for starting this! I'm really sorry that I've not had time to review this. Things are winding down with my internship so I can help get this through this week. 


I totally agree that unrelated test failures can be fixed in subsequent PR's, but there was already a typo in this PR that caused a bug, which wasn't apparent until I pulled this branch and ran the code locally. I've pointed out the typo and also proposed tests so that we can see right away if the code works as expected.

**Comment by scott-huberty** [comment]
Also if you're busy and won't have time to work on this soon, and are okay with me pushing directly, I can take a whack it it later this week!

**Comment by christian-oreilly** [comment]
Additional comment from working on the PyLossless paper: "we should clarify the 1-100 Hz bandpass filtering and notch filtering  as "flagging filters" and the raw object output by the pipeline should not be modified by them (lossless). If these filters are to be used for analysis, then they should be applied by the rejection policy. This can be dealt seamlessly by, e.g., a binary argument "apply_flagging_filters", to the rejection policy class. "

**Comment by scott-huberty** [comment]
@christian-oreilly in the interest of getting unstuck here I propose we merge this and #145 (merge in the conceptual sense, not literally a git merge). Basically, we merge this PR with the `apply` method as you wrote it, but we carry over the `RejectionPolicy` constructor I wrote in #145 (I think it's important for the `RejectionPolicy` class to have a constructor).

**Comment by christian-oreilly** [comment]
> but we carry over the RejectionPolicy constructor I wrote in https://github.com/lina-usc/pylossless/pull/145 (I think it's important for the RejectionPolicy class to have a constructor).

Not sure what you mean by this. The version of that class I wrote in this PR has a constructor. Maybe I am misunderstanding your message. Anyway, fine with me to proceed as you prefer. Unfortunately, I'll have very little time between now and the end of the month... so don't feel you need to wait after me if you are confident with your changes.

**Pull request merged**